collection-scripts/gather_sno

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+BASE_COLLECTION_PATH="must-gather"
+NODES_PATH=${OUT:-"${BASE_COLLECTION_PATH}/nodes"}
+
+function start_debug_pod_for_each_node {
+
+    #Create a path for each node
+    mkdir -p "$NODES_PATH"/"$1"
+    
+    local DEBUG_POD_NAME_PREFIX=""
+
+    #Start Debug pod force it to stay up until removed
+    oc debug node/"$1" -- chroot /host /bin/bash -c 'sleep 300' > /dev/null 2>&1 & 
+
+}
+
+function get_system_data_of_node {
+
+    #Debug pod name start with the name of the node (after removing all "." for the node name)
+    DEBUG_POD_NAME_PREFIX=${1//./}
+ 
+    counter=0
+
+    #Allow 3 sec max for the debug pod to register 
+    while [ $counter -lt 3 ]
+    do
+      #Get the debug pod that is deployed in the MG namespace on the specified node
+      debugPod=$(oc get pods -o jsonpath='{range .items[?(@.spec.nodeName=="'"$1"'")]}{.metadata.name}{"\n"}{end}' | grep "^${DEBUG_POD_NAME_PREFIX}-" | tail -n 1)
+
+      if [ -n "$debugPod" ]; then
+        break 
+      fi
+
+      counter=$((counter + 1))  
+      sleep 1  
+      
+    done
+
+    if [ -z "$debugPod" ]
+    then
+      echo "Debug pod for node ""$1"" never activated"
+    else
+      oc wait --for=condition=Ready pod/"$debugPod"  --timeout=30s
+
+      #Collect systemctl list unit files
+      echo "Collecting list-unit-files"
+      oc exec "$debugPod" -- chroot /host /bin/bash -c "systemctl list-unit-files" > "$NODES_PATH"/"$1"/list_units_files 
```

Comment:
https://github.com/sosreport/sos/blob/main/sos/report/plugins/systemd.py#L39 is how this is collected with sosreport.